### PR TITLE
CTPPS: alignment update

### PR DESCRIPTION
**Pull request by jan-kaspar**
This PR provides several fixes and improvements for CTPPS alignments. The definitive target is to have the alignments in DB, but we are too late for the deadline of 5 March. We plan to have it done for the re-reco of 2018 data.

This PR is built on top of #22264 and thus has the same dependencies, e.g. #22290. Consequently, some tests will necessarily fail before the latter is merged.

### CODE REVIEW

**Comment by jan-kaspar** [comment]
@fabferro FYI

**Comment by davidlange6** [comment]
please test

**Comment by perrotta** [comment]
@jan-kaspar : as soon as #22264 is merged in a IB, please rebase so that we can get rid of the commits which are in common and only evaluate the pure additions given by this PR

**Comment by perrotta** [comment]
@davidlange6 : this has to be tested with #22290

**Comment by perrotta** [comment]
please test with #22290

**Comment by perrotta** [comment]
@jan-kaspar : since CMSSW_10_1_X_2018-03-04-2300 the PR #22264 was included in the IB

Could you please rebase to that IB (or a newer one) so that we can get rid here of the 24 commits and the +2,224 −124 changed lines of code (still we expect some remaining +- 1200 lines of code with difference: therefore we will not get bored anyhow ;-) )

**Comment by fabferro** [comment]
@perrotta @jan-kaspar : I've rebased on the master

**Comment by davidlange6** [comment]
please test

**Comment by perrotta** [comment]
please test
(to get rid of extra commits)

**Comment by ianna** [comment]
+1

**Comment by perrotta** [comment]
+1
- Reco changes are limited to the config ```RecoCTPPS/Configuration/python/recoCTPPS_sequences_cff.py```, where the renamed and updated```ctppsIncludeAlignmentsFromXML``` (with the explicit mention of the XML in the name) is imported
- Jenkins tests pass with no differences

File: Geometry/VeryForwardGeometryBuilder/test/print_geometry_info_cfg.py [hunk 1]
```diff
@@ -11,14 +11,15 @@
...
 
 # geometry
-process.load("Geometry.VeryForwardGeometry.geometryRP_cfi")
+process.load("Geometry.VeryForwardGeometry.geometryRPFromDD_2017_cfi")
 
 # load alignment correction
-process.load("Geometry.VeryForwardGeometryBuilder.ctppsIncludeAlignments_cfi")
-process.ctppsIncludeAlignments.RealFiles = cms.vstring("Alignment/CTPPS/data/RPixGeometryCorrections.xml")
+process.load("Geometry.VeryForwardGeometryBuilder.ctppsIncludeAlignmentsFromXML_cfi")
+process.ctppsIncludeAlignmentsFromXML.RealFiles = cms.vstring("Geometry/VeryForwardGeometryBuilder/test/sample_alignment_corrections.xml")
```
Comment (arunhep) [inline Geometry/VeryForwardGeometryBuilder/test/print_geometry_info_cfg.py]: what are the plans to have this in DB?

Comment (fabferro) [reply Geometry/VeryForwardGeometryBuilder/test/print_geometry_info_cfg.py]: Actually we would have liked to have it already, but we had to postpone it due to other urgencies. We are now planning to have it for the next release, probbaly 10_2 if not before.

**Comment by fabiocos** [comment]
@fabferro what is the real time target of this PR? I understand that a DB version will not be ready before 10_2_X

**Comment by jan-kaspar** [comment]
@fabiocos If I understand correctly your question, we'd like this PR to be included in CMSSW_10_1_0.

**Comment by fabferro** [comment]
@fabiocos : I confirm @jan-kaspar 's reply

**Comment by arunhep** [comment]
+1

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
+1

**Pull request merged**